test/runtime/kafka.go

```diff
@@ -197,7 +201,12 @@ var _ = Describe("RuntimeValidatedKafka", func() {
 			"Processed a total of %d messages", MaxMessages))
 
 		By("Disable topic")
-		res = consumer(disallowTopic, MaxMessages)
-		res.ExpectFail("Kafka consumer can access to disallowTopic")
+		// Consumer timeout didn't work correctly, so make sure that AUTH is present in the reply
+		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
This does not need to be a separate background context. It will not cause an issue here hopefully,  but it was causing the earlier race we saw in `allowedTopic`. Kafka always persists the messages, so we do not need to start the consumer beforehand, and  `--from-beginning` makes sure consumer reads from start of the pipe.

Reply:
Also if it not a separate background context, we will not require `err = res.WaitUntilMatch`

Reply:
What is a "separate background context"?

The change seems to do the right thing which is to match against the output of the consumer script to see the topic unauthorized output instead of assuming the the consumer script actually fails. It may retry for quite a while.

Reply:
I mean `ctx, cancel := context.WithCancel(context.Background())` is not required. The change is not incorrect. I was pointing out that earlier we had to explicitly put a sleep 10 after starting the consumer background task, because this "background consume task "  was not ready before we start producing to it (I might not be using the "correct" golang term here ...sorry, but I hope you get what I am trying to say) 

This is working because we are using a `res.WaitUntilMatch` 
and we are consuming on a disallowed topic.

Try this scenario:

1. ctx, cancel := context.WithCancel(context.Background()) for a consumer.
2. consume on ` allowedTopic` without waiting and expect a specific output *without* using `--from-beginning`
3. Produce a set of messages 
4. This will fail if we don't had some sort of wait/sleep between steps 1,2 because the "background task" takes a while to come up. This was also the reason we had the 10 sec sleep in all our earlier bash tests.

Again, I am not a golfing pro so excuse the incorrect usage of terms.